WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverViewModel.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+import UIKit
+import WordPressUI
+
+final class CompliancePopoverViewModel: ObservableObject {
+    @Published
+    var isAnalyticsEnabled: Bool = true
+    var coordinator: CompliancePopoverCoordinator?
+
+    private let defaults: UserDefaults
+
+    init(defaults: UserDefaults) {
+        self.defaults = defaults
+    }
+
+    func didTapSettings() {
+        coordinator?.navigateToSettings()
+        defaults.shouldShowCompliancePopup = true
+    }
+
+    func didTapSave() {
+        let appAnalytics = WordPressAppDelegate.shared?.analytics
+        appAnalytics?.setUserHasOptedOut(!isAnalyticsEnabled)
+
+        let context = ContextManager.shared.mainContext
+        guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: context) else {
+            return }
```

Comment:
🟡 The `didTapSave` method is unlikely to be called from a background thread, but to be extra safe, I suggest to use this thread-safe context manager method:

```swift
let account = ContextManager.shared.performQuery { context -> WPAccount? in 
   return try? WPAccount.lookupDefaultWordPressComAccount(in: context)
}
```